File: lib/go/internal/types/InvalidMessage.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

import "fmt"

// InvalidMessage indicates that a serialized Telepact message failed validation.
type InvalidMessage struct {
	Cause error
}

// NewInvalidMessage constructs an InvalidMessage with the provided cause.
func NewInvalidMessage(cause error) *InvalidMessage {
	return &InvalidMessage{Cause: cause}
}

// Error implements the error interface.
func (e *InvalidMessage) Error() string {
	if e == nil {
		return "<nil>"
	}
	if e.Cause != nil {
		return fmt.Sprintf("invalid message: %v", e.Cause)
	}
	return "invalid message"
}

// Unwrap exposes the underlying cause for errors.Is/As.
func (e *InvalidMessage) Unwrap() error {
	if e == nil {
		return nil
	}
	return e.Cause
}

File: lib/py/telepact/internal/validation/InvalidMessage.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from typing import Optional


class InvalidMessage(Exception):
    def __init__(self, cause: Optional[Exception] = None) -> None:
        super().__init__(str(cause) if cause else None)
        self.cause = cause
